File: Exercism Codes/Python/nth_prime.py
def prime(number):
    if number <= 0:
        raise ValueError("there is no zeroth prime")
    if number == 1:
        return 2

    i, count = 1, 1

    while count != number:
        i += 2
        if is_prime(i):
            count += 1

    return i


def is_prime(number):
    return not any(number % x == 0 for x in range(3, int(number ** 0.5) + 1, 2))

File: Exercism Codes/C#/NthPrime.cs
public static class NthPrime
{
    public static int Prime(int nth)
    {
        if (nth <= 0)
        {
            throw new ArgumentOutOfRangeException(nameof(nth), "There is no zeroth prime.");
        }
        if (nth == 1)
        {
            return 2;
        }

        int i = 1;
        int count = 1;

        while (count != nth)
        {
            i += 2;
            if (isPrime(i))
            {
                count++;
            }
        }

        return i;
    }

    private static bool isPrime(int number)
    {
        for (int i = 3; i <= Math.Sqrt(number); i += 2)
        {
            if (number % i == 0)
            {
                return false;
            }
        }

        return true;
    }
}